assigner.py

```diff
@@ -1,314 +1,67 @@
-    return bapi
-
-
-def _setup_casa_api(url):
-    """
-    Setup the CASA/Biztera API object
-    """
-    capi_key = os.environ.get('CASA_API_KEY')
-
-    if (capi_key is None):
-        logger.critical("No CASA API Key passed in environment variable CASA_API_KEY")
-        sys.exit(127)
-
-    capi = casa.Casa(url=url, api_key=capi_key)
-    return capi
-
-
-def autocasa(bapi, capi, bcfg, ccfg, dry_run):
-    """
-    This will search through bugs and update CASA accordingly.
-    @bcfg: bugzilla configuration dict
-    @ccfg: casa configuration dict
-    """
-
-    bcfg_va = bcfg.get('va')
-    bcfg_rra = bcfg.get('rra')
-
-    # Look for bugs that are up to ccfg.lookup_period_in_days days old
-    lookup_period = (datetime.now() - timedelta(days=ccfg.get('lookup_period_in_days'))).strftime("%Y-%m-%dT%H:%M:%SZ")
-
-    # Look for all registered products, for that lookup_period
-    terms = [{'product': bcfg_va.get('product')}, {'product': bcfg_rra.get('product')},
-             {'component': bcfg_va.get('component')}, {'component': bcfg_rra.get('component')},
-             {'last_change_time': lookup_period},
-             {'creator': ccfg.get('bot_email')}
-            ]
-    bugs = bapi.search_bugs(terms)['bugs']
-    logger.debug('Analyzing {} bugs...'.format(len(bugs)))
-
-    for bug in bugs:
-        logger.debug("Processing bug {}...".format(bug.get('id')))
-        # Get casa project id and other metadata
-        comments = bapi.get_comments(bug.get('id'))['bugs'][str(bug.get('id'))]['comments']
-        casa_data = capi.parse_casa_comment(comments[0]['text'])
-        project_id = casa_data.get('project_id')
-        if (len(casa_data)) == 0:
-            logger.warning("Could not find any CASA data in comment 0 even thus this comment was created by CASA!")
-            continue
 
-        # STEP 1
-        # Some basic checks that we can update that project
-        ## Have bugzilla support?
-        casa_project = capi.casa_get_project(casa_data.get('project_id'))
-        if casa_project['syncedToIntegrations']['bugzilla'] is not True:
-            logger.warning('Project {} has no bugzilla integration, skipping!'.format(project_id))
-            continue
+class Assigner:
+    def __init__(self, url, service_cfg, dry_run=True):
+        self.service = service_cfg
+        self.assignees = self.service["assignees"]
+        self.api_key = "CHANGE THIS TO PARAM STORE CALL"
+        self.api = bugzilla.Bugzilla(url=url, api_key=self.api_key)
+        self.dry_run = dry_run
 
-        # Ensure this project cares about security and thus has a security tab/channel
-        if (casa_project.get('securityPrivacy') is None) or (casa_project['securityPrivacy'].get('security') is None):
-            logger.warning('Project {} has no securityPrivacy.security component, skipping!'.format(project_id))
-            continue
+    def next_assignee(self):
+        return random.choice(self.assignees)
```

Comment:
Yes, this is crud, but I have a strong sense that this will be more random than pickle queue logic (which I don't trust) and will remove state requirements entirely.

Reply:
i would disagree with this

Reply:
I will write some test code to validate my assumptions.  That said, I think the pickle strategy in Lambda will need to be restructured regardless at some level due to a lack of persistent storage unless we want to introduce databases/S3/INSERT_OTHER_TECH into the mix.

This is of course, still a work in progress and the method is abstracted, so it should be easy to functionally interate on or change strategies for.

Reply:
yup pickle cannot work in lambda - dynamodb is probably the easiest to call from there

Reply:
For sanity sake: https://gist.github.com/claudijd/15ccfe5f72841ff4bc2143728ae2ffd6